protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -54,31 +54,33 @@ impl LightNodeV1Operations for LightNodeV1 {
 
 impl LightNodeV1 {
 	pub async fn tick_block_proposer(&self) -> Result<(), anyhow::Error> {
-		let block = self.memseq.wait_for_next_block().await?;
-		match block {
-			Some(block) => {
-				info!("Built block {:?} with {:?} transactions", block.id(), block.transactions.len());
-				let block_blob = self.pass_through.create_new_celestia_blob(
-					serde_json::to_vec(&block)
-						.map_err(|e| anyhow::anyhow!("Failed to serialize block: {}", e))?,
-				)?;
-
-				let height = self.pass_through.submit_celestia_blob(block_blob).await?;
-
-				debug!("Submitted block: {:?} {:?}", block.id(), height);
-			}
-			None => {
-				// no transactions to include
+		let start_time = std::time::Instant::now();
+		let mut blocks = Vec::new();
+		while start_time.elapsed().as_millis() < 250 {
+			let block = self.memseq.wait_for_next_block().await?;
```

Comment:
A tokio::Timeout should be use here because if the `wait_for_next_bloc` lasts more than the remaining time and there's already received block, they won't be submitted in time.

Reply:
Yeah

Reply:
There's danger here with the blocks getting dropped that I'm still trying to figure out how I want to address. I may need to implement a transaction style API on the mempool.

Reply:
Removed and moved over to #258